cli/cli/.goreleaser.yml

```diff
@@ -5,24 +5,34 @@ before:
 
 builds:
   - id: cli
-    env:
-      # We need to set this to get a statically-linked binary, else we'll get a dynamically-linked binary that won't be able to be run anywhere
-      - CGO_ENABLED=0
-    main: .
     binary: "{{ .Env.CLI_BINARY_FILENAME }}"
+    main: .
     goos:
-      - linux
       - darwin
+      - linux
     goarch:
       - amd64
       - arm64
+    env:
+      - CGO_ENABLED=1
+      - CC={{ if index .Env "CC_WITH_DEFAULT" }}{{ .Env.CC_WITH_DEFAULT }}{{ else }} clang {{ end }}
```

Comment:
Using an environment variable to update the compilers for arm64. This is because I haven't figure out a way to download o64h-clang locally - need to investigate image details further to fully understand how to download - it requires (osxcross). I feel like because this is open source repo -- i am reluctant to introduce new tooling for local development. I can maybe take a look it as a FLUP ( but imo it's not high priority)

Reply:
Yeah I agree, for those things makes sure you stick comments everywhere explaining what they do and why they are needed, they look pretty cryptic but if that's what we need to get it to work, IMO no need to invest extra time just to make this cleaner